src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -705,7 +705,7 @@ private void AccumulateMatchingCommands(IEnumerable<string> commandNames)
                         moduleName = this.Module[0];
                     }
 
-                    bool isPattern = WildcardPattern.ContainsWildcardCharacters(plainCommandName);
+                    bool isPattern = WildcardPattern.MayBeWildcardPattern(plainCommandName);
```

Comment:
( Perhaps the best way would be to delegate this to a filesystem provider or even to .Net Core although it looks too difficult and big change. )